squidDrone/src/com/com_message_buffer.hpp

```diff
@@ -0,0 +1,60 @@
+#ifndef SRC_COM_MESSAGE_BUFFER_HPP_
+#define SRC_COM_MESSAGE_BUFFER_HPP_
+
+#include <array>
+#include <queue>
+#include "com_types.hpp"
+
+namespace com {
+  /** 
+    * @brief This class represents a queue style buffer for communication
+    * frames exchanged between drones.
+    * 
+    */
+  class ComMessageBuffer {
+    public:
+      /** 
+       * @brief Construct a new Com Message Buffer object
+       * 
+       */
+      ComMessageBuffer(){};
+
+      /** 
+       * @brief Transfer data to queue buffer. 
+       * 
+       * @param data Reference to data frame. Must conform to data frame specifications.
+       * @return types::ComError COM_OK if ok, COM_BUFFER_IO_ERROR if not ok.
+       * 
+       */
+      auto PutData(std::array<std::uint8_t, 32> &data) noexcept -> types::ComError;
```

Comment:
PutData first parameter "data" is an array with a missing second template parameter (32)

Reply:
I think I made a mistake here - resolved